### DISCUSSION TITLE
"AND" filter support and custom module filters extension point

### DISCUSSION THREAD

**Discussion by electroma** [pnpm/pnpm#3894] [Ideas]
We're using PNPM in the monorepo with over 150 modules. So far it worked really well.
One of our requirements is to be able to deploy to multiple target cloud environments.
And each deliverable should go to one or many cloud environments.

Our CI implementation relies on GIT filters like `...[commit-hash]...`, though now we need to add one more filter based on target environment.

The first idea was to use AND filter criteria, but it looks like it is not supported (smth like `--filter "...[hash]... AND @cloud1/*"` would do).
The alternative would be to write a plugin to implement custom selector, but it looks like there is no extension point for that.

Please suggest if there is any workaround or planned feature for it?

**Comment by zkochan** [comment]
We can allow custom filters through `.pnpmfile.cjs`

but this should already work: `--filter "...@cloud1/*[hash]..."`

**Comment by electroma** [comment]
Thank you @zkochan!
Can you please share an example of custom filter implementation via `.pnpmfile.cjs`?

**Reply by zkochan** [reply]
I cannot, such feature currently does not exist. This doesn't work? ` --filter "...@cloud1/*[hash]..."`

**Reply by electroma** [reply]
it does, though it would require us to encode "target cloud" in the module namespace/name. it could be cleaner to put it into a separate package manifest field
